nautobot/utilities/forms/widgets.py

```diff
@@ -153,6 +155,43 @@ def add_query_param(self, name, value):
 
         self.attrs[key] = json.dumps(values, ensure_ascii=False)
 
+    def get_context(self, name, value, attrs):
+
+        # This adds null options to DynamicModelMultipleChoiceField selected choices
+        # example <select ..>
+        #           <option .. selected value="null">None</option>
+        #           <option .. selected value="1234-455...">Rack 001</option>
+        #           <option .. value="1234-455...">Rack 002</option>
+        #          </select>
+        # Prepend null choice to self.choices if
+        # 1. form field allow null_option e.g. DynamicModelMultipleChoiceField(..., null_option="None"..)
+        # 2. if null is part of url query parameter for name(field_name) i.e. http://.../?rack_id=null
+        # 3. if both value and choices are iterable
+        if (
+            self.attrs.get("data-null-option")
+            and isinstance(value, Iterable)
```

Comment:
Caution here - a `str` is an `Iterable` - might be safer to specifically check for `(list, tuple)` here and below instead?

Reply:
Below `self.choices` is a generator not list or tuple